sqlglot/dialects/tsql.py

```diff
@@ -186,7 +186,7 @@ class Parser(Parser):
         def _parse_convert(self, strict):
             to = self._parse_types()
             self._match(TokenType.COMMA)
-            this = self._parse_field()
+            this = self._parse_column()
```

Comment:
Let's change the base parser to do the same thing.

Reply:
Good thinking